node/utils/metrics/organization.ts

```diff
@@ -0,0 +1,96 @@
+import type { Logger } from '@vtex/api/lib/service/logger/logger'
+
+import type { Metric } from './metrics'
+import { sendMetric } from './metrics'
+
+interface UpdateOrganizationFieldsMetric {
+  update_details: { properties: string[] }
+}
+
+type UpdateOrganization = Metric & {
+  fields: UpdateOrganizationFieldsMetric
+}
+
+export interface UpdateOrganizationParams {
+  account: string
+  currentOrganizationData: Organization
+  updatedProperties: UpdatedOrganizationProps
+}
+
+const buildUpdateOrganizationMetric = (
+  account: string,
+  updatedProperties: string[]
+): UpdateOrganization => {
+  const updateOrganizationFields: UpdateOrganizationFieldsMetric = {
+    update_details: { properties: updatedProperties },
+  }
+
+  return {
+    account,
+    description: 'Update Organization Action - Graphql',
+    fields: updateOrganizationFields,
+    kind: 'update-organization-graphql-event',
+    name: 'b2b-suite-buyerorg-data',
+  } as UpdateOrganization
+}
+
+const getFieldsNamesByFieldsUpdated = (
+  updateOrganizationParams: UpdateOrganizationParams
+): string[] => {
+  const updatedPropName: string[] = []
+
+  const { currentOrganizationData, updatedProperties } =
+    updateOrganizationParams
+
+  for (const key in updatedProperties) {
+    if (Object.prototype.hasOwnProperty.call(updatedProperties, key)) {
```

Comment:
Since we're already looping through the keys within `updatedProperties`, this call shouldn't be necessary.

Also, when it is necessary, you should be able to call the function directly on the object, like this:
```suggestion
    if (updatedProperties.hasOwnProperty(key)) {
```

Reply:
I tried to use it, but the lint suggested the change. Can I keep this eslint error?
![image](https://github.com/vtex-apps/b2b-organizations-graphql/assets/5332361/8a714e6a-0db2-4565-8537-0ad5648961ef)

Reply:
Got it, so I think it's suggesting the use of `Object.hasOwnProperty(updatedProperties, key)`. I just wouldn't use `Object.prototype` here, and also not use `.call(`, for example.

Reply:
But I still think this line shouldn't be necessary. Fixing the type of the updated fields to be based on Organization will probably remove any lint requesting this check to be made.